### PR TITLE
Add MP JWT 1.2 Release notes

### PR DESCRIPTION
**Pull request by sberyozkin**
This PR add MP JWT 1.2 Release notes following the format used by MP Config, OpenApi etc.

I'll appreciate a quick review as I'd like to do the `RC1` release asap

CC @Emily-Jiang

### CODE REVIEW

**Comment by sberyozkin** [comment]
@radcortez Thanks, I'll go ahead with the merge a bit later today in case someone else proposes to tweak it somehow, but yeah, I think it captures the main changes well enough.

**Comment by sberyozkin** [comment]
Fixed a small typo

File: spec/src/main/asciidoc/release-notes.asciidoc [hunk 1]
```diff
@@ -14,20 +14,45 @@
...
-## Release Notes
+[[release_notes_12]]
+== Release Notes for MicroProfile JWT 1.2
 
-### Changes in 1.1.1-RC2
+A full list of changes delivered in the 1.2 release can be found at link:https://github.com/eclipse/microprofile-jwt-auth/milestone/7[MicroProfile JWT 1.2 Milestone].
+
+=== API Changes
+
+- Convenience method has been added to allow retrieving claims from `JsonWebToken` by using the `Claims` enum (link:https://github.com/eclipse/microprofile-jwt-auth/issues/154[#154])
```
Comment (Emily-Jiang) [inline spec/src/main/asciidoc/release-notes.asciidoc:24]: A convenience

File: spec/src/main/asciidoc/release-notes.asciidoc [hunk 2]
```diff
@@ -14,20 +14,45 @@
...
+
+=== Spec Changes
+
```
Comment (Emily-Jiang) [inline spec/src/main/asciidoc/release-notes.asciidoc:27]: Can you also add the link to the issue for the following items?

File: spec/src/main/asciidoc/release-notes.asciidoc (hunk 1 above)
Comment (sberyozkin) [reply spec/src/main/asciidoc/release-notes.asciidoc]: Done

File: spec/src/main/asciidoc/release-notes.asciidoc (hunk 2 above)
Comment (sberyozkin) [reply spec/src/main/asciidoc/release-notes.asciidoc:27]: @Emily-Jiang thanks, added and verified the links

**Comment by sberyozkin** [comment]
Hi All, merging now, thanks

**Pull request merged**